contracts/cw-ics721-bridge/src/ibc_helpers.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{
+    ibc::{NonFungibleTokenPacketData, IBC_VERSION},
+    ContractError,
+};
+
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, IbcAcknowledgement, IbcChannel, IbcEndpoint, IbcOrder,
+    StdError, StdResult,
+};
+
+/// Tries to remove the source prefix from a given class_id. If the
+/// class_id does not begin with the given prefix, returns
+/// `None`. Otherwise, returns `Some(unprefixed)`.
+pub(crate) fn try_pop_source_prefix<'a>(
+    source: &IbcEndpoint,
+    class_id: &'a str,
+) -> Option<&'a str> {
+    let source_prefix = get_endpoint_prefix(source);
+    // This must not panic in the face of non-ascii, or empty
+    // strings. We can not trust classID as it comes from an external
+    // IBC connection.
+    class_id.strip_prefix(&source_prefix)
+}
```

Comment:
Something like this I would almost want in a package? :thinking:

A lot of these ibc helpers are pretty general. IIRC @shanev has suggested they should be in a cw package before.

Reply:
Guess they are fine here for now unless others have a better idea or suggestion.

Reply:
I'm not opposed to doing that. We could set up a similar structure to dao-contracts and create a `packages` folder.

The functions in here are pretty ICS 721 specific. Do you imagine other implementations wanting to pull these in?

Reply:
> I'm not opposed to doing that. We could set up a similar structure to dao-contracts and create a `packages` folder.
> 
> The functions in here are pretty ICS 721 specific. Do you imagine other implementations wanting to pull these in?

Let's not over-optimize, can pull them into a package if need be in a separate PR.